apps/previewgenerator.sh

```diff
@@ -16,67 +16,218 @@ debug_mode
+            ;;
+
+            MP3)
+                clear
+                occ_command config:system:set enabledPreviewProviders 6 --value="OC\\Preview\\MP3"
+            ;;
+
+            TXT)
+                clear
+                occ_command config:system:set enabledPreviewProviders 7 --value="OC\\Preview\\TXT"
+            ;;
+
+            Movie)
+                clear
+                occ_command config:system:set enabledPreviewProviders 9 --value="OC\\Preview\\Movie"
+            ;;
+
+            *)
+            ;;
+        esac
+    done 9< results
+    rm -f results
+    clear
+fi
+
+# Enable additional previews (most likely not working; remove the # to enable the specific preview)
+#occ_command config:system:set preview_libreoffice_path --value="/usr/bin/libreoffice"
+#occ_command config:system:set enabledPreviewProviders 4 --value="OC\\Preview\\XBitmap"
+#occ_command config:system:set enabledPreviewProviders 8 --value="OC\\Preview\\Illustrator"
+#occ_command config:system:set enabledPreviewProviders 10 --value="OC\\Preview\\MSOffice2003"
+#occ_command config:system:set enabledPreviewProviders 11 --value="OC\\Preview\\MSOffice2007"
+#occ_command config:system:set enabledPreviewProviders 12 --value="OC\\Preview\\MSOfficeDoc"
+#occ_command config:system:set enabledPreviewProviders 13 --value="OC\\Preview\\OpenDocument"
+#occ_command config:system:set enabledPreviewProviders 14 --value="OC\\Preview\\PDF"
+#occ_command config:system:set enabledPreviewProviders 16 --value="OC\\Preview\\Postscript"
+#occ_command config:system:set enabledPreviewProviders 17 --value="OC\\Preview\\StarOffice"
+#occ_command config:system:set enabledPreviewProviders 20 --value="OC\\Preview\\Font"
+#occ_command config:system:set enabledPreviewProviders 21 --value="OC\\Preview\\HEIC"
+
+# check if the previewgenerator is installed and enabled
+if [ -d "$NC_APPS_PATH/previewgenerator" ]
+then
     # Set aspect ratio
-    occ_command config:app:set --value="32 64 1024"  previewgenerator squareSizes
-    occ_command config:app:set --value="64 128 1024" previewgenerator widthSizes
-    occ_command config:app:set --value="64 256 1024" previewgenerator heightSizes
+    occ_command config:app:set previewgenerator squareSizes --value="32 256"
+    occ_command config:app:set previewgenerator widthSizes  --value="256 384"
+    occ_command config:app:set previewgenerator heightSizes --value="256"
+    occ_command config:system:set preview_max_x --value="2048"
+    occ_command config:system:set preview_max_y --value="2048"
+    occ_command config:system:set jpeg_quality --value="60"
+    occ_command config:app:set preview jpeg_quality --value="60"
     
-    # Add crotab
-    crontab -u www-data -l | { cat; echo "@daily php -f $NCPATH/occ preview:pre-generate >> /var/log/previewgenerator.log"; } | crontab -u www-data -
+    msg_box "In the last step you can define a nextcloud-user for which you want to run the preview-generation. "
```

Comment:
Please remove this option. If the user makes a typo, the pre-generate will never happen. Also, as it's a cronjob, it will run in the background and no one will notice as it's in the middle of the night.

If you *really* think this option is useful and want to add this feature, please make `$nextcloud_user` `www-data` per default, and rename the variable to `cronjob_user`. So that if nothing is chosen, use www-data.

Something like: "The default user for generating previews are www-data. There's no reason to change that expect in special edge cases. Please only change this if you know what you are doing!"

Adding complexity = more likely to get bugs. Keep it simple. :)

Reply:
Also, if you go for the second option, add a check with `occ_command user:list | grep $user` to check if the user exists. If not, then ask for input again.

Thanks.

Reply:
The thing is, that I don't want to change the cronjob-user but define a nextcloud-user for which I want to run the command as I explained here: https://github.com/nextcloud/vm/pull/1002#issuecomment-569841096
But I added the check. Thanks for the advice.

Reply:
It's not what you want, it's what everyone wants. :) This VM is downloaded several thousand times. Having a setting that can be confusing (e.g. why even run with a certain user? Not everyone have the same setup as you.) is asking for issue reports when users say: "I installed preview generator, but it doesn't work".

Please change so that `www-data` is default, add text to explain what happens if you change it, like you explained for me. www-data is default for every user, having a single user is *very much an edge case* which involves group folders and so on.

Reply:
actually I don't know what you mean. The system user is actually set to www-data
and if not set the preview generation runs for all nextcloud-users as default.

look at line 243:
`crontab -u www-data -l | { cat; echo "0 4 * * * php -f $NCPATH/occ preview:pre-generate $nextcloud_user >> /var/log/previewgenerator.log"; } | crontab -u www-data -`

so if the $nextcloud_user variable is empty, nothing gets added. So then it runs for all nextcloud-users.

Reply:
I have tried to improve the explanation of this a bit. and make sure that it runs for all nextcloud-users, if no user is set up